app/components/documents/document-card.js

```diff
@@ -125,12 +127,26 @@ export default class DocumentsDocumentCardComponent extends Component {
         'filter[:id:]': id,
         include: 'document-container,document-container.type,access-level',
       });
-
+    const loadPiecePart = (id) => 
+      this.store.queryOne('piece-part', {
+        'filter[report][:id:]': id,
+      });
     if (this.args.piece) {
       this.piece = this.args.piece; // Assign what we already have, so that can be rendered already
       this.piece = yield loadPiece(this.piece.id);
       this.documentContainer = yield this.piece.documentContainer;
       yield this.loadVersionHistory.perform();
+      if (isPresent(this.args.label)) {
+        this.label = this.args.label;
+      } else {
+        const piecePart = yield loadPiecePart(this.piece.id);
```

Comment:
This will run on every document-card.
Even on those were we know in advance that there will never be a piece-part (like agendaitem documents)

Assuming we have to query twice (once for report, once for minutes) this will increase the general query load.
Assuming only pieces that are generated are of subtype "report" or "minutes"  may have pieceparts.
Perhaps we could check that? 
`this.args.piece.constructor.modelName === 'report' || this.args.piece.constructor.modelName === 'minutes'` > 
check for pieceParts
In case of `report`, legacy decision reports have the type "report" but have no pieceParts so extra query is needed
In case of `minutes`, there is no legacy so can we assume that if we find that type it will always be generated? if so we could skip the extra query here.